atomic_reactor/plugins/pre_add_filesystem.py

```diff
@@ -0,0 +1,209 @@
+                              for task completion
+        """
+        # call parent constructor
+        super(AddFilesystemPlugin, self).__init__(tasker, workflow)
+        self.koji_hub = koji_hub
+        self.koji_auth_info = {
+            'proxyuser': koji_proxyuser,
+            'ssl_certs_dir': koji_ssl_certs_dir,
+            'krb_principal': koji_krb_principal,
+            'krb_keytab': koji_krb_keytab,
+        }
+        self.from_task_id = from_task_id
+        self.poll_interval = poll_interval
+
+    def is_kickstart_type(self, base_image):
+        return base_image.strip().lower() == 'scratch:kickstart'
+
+    def parse_image_build_config(self, config_file_name):
+
+        # Logic taken from koji.cli.koji.handle_image_build.
+        # Unable to re-use koji's code because "cli" is not
+        # a package of koji and this logic is intermingled
+        # with CLI specific instructions.
+
+        args = []
+        opts = {}
+
+        config = ConfigParser()
+        config.read(config_file_name)
+
+        image_name = None
+
+        section = 'image-build'
+        for option in ('name', 'version', 'arches', 'target', 'install_tree'):
+            value = config.get(section, option)
+            if option == 'arches':
+                value = [arch for arch in value.split(',') if arch]
+            elif option == 'name':
+                image_name = value
+            args.append(value)
+            config.remove_option(section, option)
+
+        for option, value in config.items(section):
+            if option in ('repo', 'format'):
+                value = [v for v in value.split(',') if v]
+            elif option in ('disk_size'):
+                value = int(value)
+            opts[option] = value
+
+        section = 'ova-options'
+        if config.has_section(section):
+            ova = []
+            for k, v in config.items(section):
+                ova.append('{}={}'.format(k, v))
+            opts['ova_option'] = ova
+
+        section = 'factory-parameters'
+        if config.has_section(section):
+            factory = []
+            for option, value in config.items(section):
+                factory.append((option, value))
+            opts['factory_parameter'] = factory
+
+        # Set some defaults.
+        opts.setdefault('disk_size', 10)
+
+        return image_name, args, {'opts': opts}
+
+    def build_filesystem(self):
+        # Kickstart file should be in the same folder as Dockerfile
+        df_path, df_dir = self.workflow.source.get_dockerfile_path()
+        kickstart = os.path.join(df_dir, 'kickstart.conf')
+        if not os.path.exists(kickstart):
+            raise RuntimeError('Kickstart file not found: {}'
+                               .format(kickstart))
+
+        image_name, args, kwargs = self.parse_image_build_config(kickstart)
+        pattern = ('{}.*(\.tar|\.tar\.gz|\.tgz|\.bzip|\.tar\.xz|\.txz)$'
```

Comment:
OK. I don't think I have ever seen a filename with a `.bzip` (rather than eg `.bz2`) extension, I don't think that's a real thing :-)

I can't imagine a scenario in which we will have asked Koji to create a filesystem image, and it wouldn't use the traditional UNIX ".tar.{compression}" scheme for naming the file.